### PR TITLE
Dump operator names of a script module, v1.4.0 pick request

### PR DESCRIPTION
**Pull request by iseeyuan**
Summary:

Introduce function jit.export_opnames(module), which returns a list of all operator names used in the module and its submodules. One usage is to have mobile custom build to link only operators in the returned list to save the mobile size.

Example:
import torch
m = torch.jit.load("example.pt")
print(torch.jit.export_opnames(m))

The outputs are in alphabetical order:
['aten::_convolution', 'aten::add.Tensor', 'aten::add_.Tensor', 'aten::addmm', 'aten::append.Tensor', 'aten::cat', 'aten::dropout', 'aten::embedding', 'aten::matmul', 'aten::max.dim', 'aten::mul.Tensor', 'aten::permute', 'aten::relu', 'aten::t', 'aten::tanh', 'prim::ListConstruct', 'prim::TupleConstruct', 'prim::TupleUnpack']

### CODE REVIEW

**Comment by gchanan** [comment]
I have a couple of concerns with this PR:
1) There are no tests
2) export_opnames is being exposed at the top-level torch namespace (`torch.export_opnames`).  This doesn't look like it's unique to this PR, as there is also `torch.import_ir_module_from_buffer`, `torch.merge_type_from_type_comment`, but it would be good not to release like this.

**Comment by gchanan** [comment]
CC @ZolotukhinM, who reviewed the original PR.

**Comment by iseeyuan** [comment]
@gchanan 
1. For tests, @ZolotukhinM did remind me after the original diff was landed to master. I'm adding them. 
2. Please let me know if there's any alternative API to have similar functionality. It may make sense to add the tests when the API concern is addressed. 
Since it's the first step for custom build, hope it could be included in 1.4. cc @ljk53 and @dreiss

**Comment by ZolotukhinM** [comment]
I think it would make sense to move it to `torch.jit.` namespace. I'd also suggest consulting with @suo or @zdevito, maybe there is a better place for it. I agree that having it in the top-level namespace is a bit strange.

**Comment by iseeyuan** [comment]
Just confirmed with @suo and he's fine with `torch.jit.export_opnames`. I'll make the change accordingly.

**Comment by iseeyuan** [comment]
> I have a couple of concerns with this PR:
> 
> 1. There are no tests
> 2. export_opnames is being exposed at the top-level torch namespace (`torch.export_opnames`).  This doesn't look like it's unique to this PR, as there is also `torch.import_ir_module_from_buffer`, `torch.merge_type_from_type_comment`, but it would be good not to release like this.

@gchanan For your second concern, export_opnames is already in torch.jit namespace (not in the top torch namespace), or did I miss anything?

**Comment by gchanan** [comment]
@iseeyuan: I believe it's in both.  Basically, everything in `torch._C.*` gets exported to `torch.*`.  And then you have a separate definition torch.jit.

So, the easiest change would just be to rename the thing in _C to `_export_names` or something (because we reserve underscore prefixed things as being private).  There might be a better way, e.g. for `torch.nn` we have `torch._C._nn` but I don't know if the JIT has that.  Maybe @suo can weigh in.

**Comment by kostmo** [comment]
## :pill: CircleCI build failures summary and remediations
As of commit a21028a:
* **1/1** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=a21028a8a2176d2ee38f1b37e2c3623c7c46cf41).

### :detective: 1 new failure recognized by patterns
The following build failures don't appear to be due to upstream breakage:
#### [See CircleCI build](https://circleci.com/gh/pytorch/pytorch/4011210) pytorch_linux_xenial_py3_6_gcc5_4_build (1/1)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=77118965) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=77118965))
<details>
<summary>
<code>Automatic merge failed; fix conflicts and then commit the result.</code>
</summary>

```
CONFLICT (add/add): Merge conflict in .circleci/docker/common/install_cache.sh 
Auto-merging .circleci/docker/common/install_cache.sh 
CONFLICT (add/add): Merge conflict in .circleci/docker/common/install_android.sh 
Auto-merging .circleci/docker/common/install_android.sh 
CONFLICT (add/add): Merge conflict in .circleci/docker/build.sh 
Auto-merging .circleci/docker/build.sh 
CONFLICT (add/add): Merge conflict in .circleci/config.yml 
Auto-merging .circleci/config.yml 
CONFLICT (add/add): Merge conflict in .circleci/cimodel/data/binary_build_definitions.py 
Auto-merging .circleci/cimodel/data/binary_build_definitions.py 
Automatic merge failed; fix conflicts and then commit the result. 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

**Review by gchanan — APPROVED** [review]
